services/core-api/app/api/mines/tailings/resources/tailings.py

```diff
@@ -113,26 +128,52 @@ def put(self, mine_guid, mine_tailings_storage_facility_guid):
         if not mine:
             raise NotFound('Mine not found.')
 
-        mine_tsf = MineTailingsStorageFacility.find_by_tsf_guid(mine_tailings_storage_facility_guid)
+        mine_tsf: MineTailingsStorageFacility = MineTailingsStorageFacility.find_by_tsf_guid(mine_tailings_storage_facility_guid)
 
         if not mine_tsf:
             raise NotFound("Tailing Storage Facility not found")
 
         data = self.parser.parse_args()
+
+        is_submitting = data.get('is_submitting')
         eor_party_guid = data.get('eor_party_guid')
-        if eor_party_guid != None and (mine_tsf.engineer_of_record is None
-                                       or eor_party_guid != mine_tsf.engineer_of_record.party_guid):
+        has_new_eor = eor_party_guid != None and (mine_tsf.engineer_of_record is None or eor_party_guid != mine_tsf.engineer_of_record.party_guid)
+        if has_new_eor:
             if mine_tsf.engineer_of_record:
                 mine_tsf.engineer_of_record.end_date = datetime.now(tz=timezone.utc)
 
             if is_minespace_user():
                 # EORs created through minespace should have a status of "pending"
                 new_status = MinePartyAppointmentStatus.pending
                 mine_party_acknowledgement_status = MinePartyAcknowledgedStatus.not_acknowledged
+
+                if is_feature_enabled(Feature.TSF_V2):
+                    if is_submitting:
+                        calculated_due_date = datetime.now() + timedelta(hours=72)
+                        mine_report_definition = MineReportDefinition.find_one_by_section('10', '4', '1', '2')
```

Comment:
Was trying to find the best way to find these specific reports that wouldn't be dependent on the environment.  I don't love hard coding the section values here, but it seemed like the most stable element of a report definition (compliance_article) to search by.

Open to ideas for a better way to handle it though.

Reply:
Oof. Yeah ngl this is gross. But then I looked at prod mine_report_definitions and there is literally "Engineer of Record (confirmation)" and "Engineer of Record (departure)" so the other option is to hardcode the report name and that is just as gross. I have no suggestions.

Reply:
Making a ticket to improve this and make it more reliable/maintainable.